### PR TITLE
Add warning while untagging an image podman-load

### LINKED ISSUE #2821: BUG: loading multiple images corrupts tags of existing ones
[Now that I have access to the nightly ppa's (cf. containers/libpod#2250), here's some more things I found]

Sometimes it's necessary to transfer images in bulk (say, to an environment that has restricted internet access). Docker supports saving/loading multiple images, and nominally, podman does too. Loading multiple images saved by docker fails (xref #2822), while loading multiple images saved by podman go through but are either loading incorrectly or even corrupting tags that might have been correct before.

This is a much subtler issue and thus harder to detect.

Steps to reproduce:
```
$ sudo podman pull ubuntu:bionic
$ sudo podman pull ubuntu:cosmic
```
Then, things look as follows.
multi1

Saving and reloading corrupts the tags.
```
$ sudo podman save docker.io/library/ubuntu:bionic docker.io/library/ubuntu:cosmic > podman_test.tar
$ sudo podman load -i podman_test.tar
```
In particular, `docker.io/library/ubuntu:cosmic` gets wrongly tagged to the commit of `docker.io/library/ubuntu:bionic`, while the actual commit of `docker.io/library/ubuntu:cosmic` gets untagged. 
multi2

Side note: in contrast to docker, there is no auto-completion for file-names after `sudo podman load -i`

### PR DESCRIPTION
**Pull request by weirdwiz**
Adds a warning if another image with the same name exists and is going to be untagged

fixes #2821

### CODE REVIEW

**Comment by baude** [comment]
bot, test pull request

File: libpod/image/pull.go [hunk 1]
```diff
@@ -149,6 +149,13 @@ func (ir *Runtime) pullGoalFromImageReference(ctx context.Context, srcRef types.
 		// Need to load in all the repo tags from the manifest
 		res := []pullRefPair{}
 		for _, dst := range manifest[0].RepoTags {
+			//check if image exists and gives a warning of untagging
+			localImage, err := ir.NewFromLocal(dst)
+			imageID := strings.TrimSuffix(manifest[0].Config, ".json")
+			if err == nil && imageID != localImage.ID() {
+				logrus.Errorf("the image %s already exists, renaming the old one with ID %s to empty string", dst, localImage.ID())
```
Comment (mheon) [inline libpod/image/pull.go]: Suggest rewording - `"tag %s already exists, untagging but not deleting old image"

Comment (mheon) [reply libpod/image/pull.go]: Oh, but you'd want to keep the ID in there... maybe `tag %s already exists as ID %s, untagging but not deleting old image`?

Comment (TomSweeneyRedHat) [inline libpod/image/pull.go]: "renaming" implies the software is in the process of changing the images name which doesn't seem to be happening?
Perhaps:
"the image %s already exists, rename the image with ID %s to another value and retry"

There's probably a better way to say the above, but that's all I got after one cup of tea.

Comment (weirdwiz) [reply libpod/image/pull.go]: Docker only gives this warning and continues to untag it, so I only added the warning

Comment (weirdwiz) [reply libpod/image/pull.go]: untagging would mean removing the tag, maybe I could reword it to 
`tag %s already exists as ID %s, tagging the old image as an empty string`

Comment (TomSweeneyRedHat) [reply libpod/image/pull.go]: Back to my original point, unless I'm being oblivious, this code is not actually tagging the old image as an empty string is it?  I think in this message you need to tell the user what action they need to do as we're not doing it for them.

Comment (mheon) [reply libpod/image/pull.go]: We're specifically retagging as empty string? I think internally that's considered an untag of the old image... @baude that sound right to you?

Comment (weirdwiz) [reply libpod/image/pull.go]: Internally this set's the name of the image  `image.Names = stringSliceWithoutValue(image.Names, name)` in the file https://github.com/containers/storage/blob/master/images.go .

The thing that is confusing me is that if I do `podman rmi imagename` then if the image exists with another tag then it simply gets untagged and is not visible when I run `podman images`, but in this case, the image is visible in `podman images` as `<none>` .

Comment (mheon) [reply libpod/image/pull.go]: ...Could this be the dreaded literal <none> tag returning... Let me poke around

**Comment by rhatdan** [comment]
@mheon @TomSweeneyRedHat @baude Where do we stand on this PR?

**Comment by mheon** [comment]
I'll give it another look, but we were just nitpicking the wording of the
warning. It's basically ready.

**Comment by TomSweeneyRedHat** [comment]
The error message still needs to change IMO.  The message implies that the code is changing something when AFAIKS, it's only reporting an issue.  The code other than the message is fine.

**Comment by weirdwiz** [comment]
@TomSweeneyRedHat AFAIK the code does remove the name of the image with the same name but a different ID as the image loaded. I am unsure whether it should be called untagging, or renaming to something empty.

**Comment by TomSweeneyRedHat** [comment]
If @mheon is happy with the messaging, then it's fine by me,  But unless I'm missing the functionality of the code (very possible), I think it's a misleading message.
